marlowe-playground-client/test/Marlowe/LintTests.purs

```diff
@@ -249,6 +251,16 @@ nonPositivePay = testWarningSimple (payContract "(Constant 0)") "The contract ca
 negativePay :: Test
 negativePay = testWarningSimple (payContract "(Constant -1)") "The contract can make a non-positive payment"
 
+payBeforeWarning :: Test
+payBeforeWarning = testWarningSimple contract "The contract makes a payment to account (AccountId 0 (Role \"role\")) before a deposit has been made"
+  where
+  contract = "When [Case (Deposit (AccountId 1 (Role \"role\") ) (Role \"role\") (Token \"\" \"\") (Constant 100)) (Pay (AccountId 0 (Role \"role\")) (Party (Role \"role\")) (Token \"\" \"\") (Constant 1) Close)] 10 Close"
+
```

Comment:
Here is a test that should produce a warning and I guess it does not with current implementation:
```When [Case (Deposit (AccountId 1 (Role "role")) (Role "role") (Token "" "") (Constant 10)) Close] 2 (Pay (AccountId 1 (Role "role")) (Party (Role "role")) (Token "" "") (Constant 10) Close)```

Reply:
indeed this test fails to produce a warning